src/gui/folder.cpp

```diff
@@ -108,15 +110,62 @@ Folder::Folder(const FolderDefinition &definition,
     _scheduleSelfTimer.setInterval(SyncEngine::minimumFileAgeForUpload);
     connect(&_scheduleSelfTimer, &QTimer::timeout,
         this, &Folder::slotScheduleThisFolder);
+
+    //initialize wamp connection
+    QWamp::Session *session;
+    _webSocket.reset(new QTcpSocket());
+//    QByteArray wsUrl = qgetenv("OWNCLOUD_WEBSOCKET_URL");
+    QByteArray wsUrl = "ws://localhost:8080/ws";
+    QObject::connect(_webSocket.data(), &QTcpSocket::connected, [&]() {
+
+        QString sessionName("sessionName");
+        session = new QWamp::Session(sessionName, *_webSocket.data(), QWamp::Session::MessageFormat::Msgpack, true);
+
+        QObject::connect(session, &QWamp::Session::joined, [&](qint64 s) {
+          qDebug() << "Session joined to realm1 with session ID " << s;
+
+          session->subscribe("etag-changed-channel", [&](const QVariantList& args, const QVariantMap& options){
+              qDebug() << "Event received";
```

Comment:
While in general I think the lambdas are cool, we should not become a nodejs with 3x nested callbacks :-)